cedar-drt/fuzz/src/lib.rs

```diff
@@ -33,180 +31,158 @@ use cedar_policy_core::extensions::Extensions;
 pub use cedar_policy_validator::{ValidationErrorKind, ValidationMode, Validator, ValidatorSchema};
 use log::info;
 
-pub struct DifferentialTester<'e> {
-    /// Rust engine instance
-    authorizer: Authorizer,
-    /// Definitional engine instance
-    def_engine: DefinitionalEngine<'e>,
-    /// Definitional validator instance
-    def_validator: DefinitionalValidator<'e>,
+/// Compare the behavior of the evaluator in cedar-policy against a custom Cedar
+/// implementation. Panics if the two do not agree. `expr` is the expression to
+/// evaluate and `request` and `entities` are used to populate the evaluator.
+pub fn run_eval_test(
+    custom_impl: &dyn CedarTestImplementation,
+    request: &ast::Request,
+    expr: &Expr,
+    entities: &Entities,
+    enable_extensions: bool,
+) {
+    let exts = if enable_extensions {
+        Extensions::all_available()
+    } else {
+        Extensions::none()
+    };
+    let eval = match Evaluator::new(request, entities, &exts) {
+        Ok(e) => e,
+        Err(_) => return, // FOR NOW just ignore errors in the restricted exprs
+    };
+    let v = match eval.interpret(expr, &std::collections::HashMap::default()) {
+        Ok(v) => Some(v),
+        Err(e) if matches!(e.error_kind(), EvaluationErrorKind::IntegerOverflow(_)) => {
+            // ignore the input if it results in an integer overflow error
+            return;
+        }
+        Err(_) => None,
+    };
+    // custom_impl.eval() returns true when the result of evaluating expr
+    // matches the expected value v
+    assert!(custom_impl.interpret(request, entities, expr, v))
```

Comment:
the comment is about `custom_impl.eval()`, but the code uses `custom_impl.interpret()`